File: dags/test_minio.js
const { Client: MinioClient } = require('minio');

const minioClient = new MinioClient({
    endPoint: '127.0.0.1',
    port: 9000,
    useSSL: false,
    accessKey: 'minioadmin',
    secretKey: 'minioadmin'
});

async function testConnection() {
    try {
        const buckets = await minioClient.listBuckets();
        console.log('Connected to MinIO! Buckets:', buckets);
    } catch (err) {
        console.error('MinIO connection failed:', err);
    }
}

testConnection();

File: dags/test_minio.py
        client = Minio(
            endpoint='127.0.0.1:9000',
            access_key='minioadmin',
            secret_key='minioadmin',
            secure=False
        )
        # Upload file
        result = client.fput_object(
            bucket_name,
            object_name,
            file_path,
        )
        print(f"Successfully uploaded {result.object_name} of size {result.size} bytes")
        return True
    except Exception as e:
        print(f"Error in upload_to_minio: {str(e)}")
        raise e
EOL